### ISSUE TITLE
Test failure on Fedora linux

### ISSUE THREAD

**Issue by bkmgit** [tigerbeetle/tigerbeetle#906]
When building on Fedora linux 39, get a test failure:
```
102/186 vsr.replica_test.test "Cluster: recovery: recovering_head, outdated start view"... Segmentation fault at address 0x0
/builddir/build/BUILD/tigerbeetle-0.13.36/src/vsr/replica_test.zig:1181:70: 0x46028c in vsr.replica_test.TestReplicas.peer_paths (test)
                    paths.appendAssumeCapacity(.{ .source = process_b.*, .target = process_a });
                                                                     ^
error: the following test command crashed:
/builddir/build/BUILD/tigerbeetle-0.13.36/zig-cache/o/6372e2f7ee0c1720263c07214eb04d14/test /usr/bin/zig
The following command exited with error code 1:
/usr/bin/zig test /builddir/build/BUILD/tigerbeetle-0.13.36/src/unit_tests.zig -lc -OReleaseSafe --cache-dir /builddir/build/BUILD/tigerbeetle-0.13.36/zig-cache --global-cache-dir /builddir/.cache/zig --name test -target native-native -mcpu x86_64 --pkg-begin vsr_options /builddir/build/BUILD/tigerbeetle-0.13.36/zig-cache/options/vBjiiAWQ0Kc_eoElcIITvwW2HC2uCfsCS5radwlEyyxAno8JAoLEW_80-w2CX984 --pkg-end -I /builddir/build/BUILD/tigerbeetle-0.13.36/src/clients/c 
error: the following build command failed with exit code 1:
zig-cache/o/d4dbb3b79804f75cb302271fbb1886a8/build /usr/bin/zig /builddir/build/BUILD/tigerbeetle-0.13.36 zig-cache /builddir/.cache/zig --verbose -Dtarget=native -Dcpu=baseline -Drelease-safe test
error: Bad exit status from /var/tmp/rpm-tmp.FznaJt (%check)
RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.FznaJt (%check)
Child return code was: 1
```
Full log https://copr-be.cloud.fedoraproject.org/results/fed500/tigerbeetle/fedora-rawhide-x86_64/06108769-tigerbeetle/build.log.gz

**Comment by sentientwaffle** [comment]
Thanks for reporting!

Unfortunately this is a miscompilation bug. Luckily this bug is only impacting the tests, not the client or replica.

As a workaround (until TigerBeetle upgrades to a newer zig version), remove `-Drelease-safe` from the `zig build ... test` command (to build the tests in `Debug` mode). This works around the compiler bug (at least on my machine).

That is, instead of:

```
/usr/bin/zig build --verbose -Dtarget=native -Dcpu=baseline -Drelease-safe --cache-dir zig-cache test
```

run

```
/usr/bin/zig build --verbose -Dtarget=native -Dcpu=baseline --cache-dir zig-cache test
```

---

Aside: I confirmed that this bug is still present in zig `0.10.0-dev.2859+0101a5f75`, but is fixed in `0.10.1` and `0.11.0-dev.3384+00ff65357`.

Minimal test case:

```zig
const std = @import("std");

pub fn main() void {
    var ab = AB.a;

    var list = List{.buffer = undefined};
    list.appendAssumeCapacity(.{ .small = 0 });
    list.appendAssumeCapacity(.{ .small = 1 });

    var count: usize = 0;
    for (list.constSlice()) |_| {
        if (ab == .a) {
            count += 1;
        }
    }
}

const AB = enum {a, b};
const List = std.BoundedArray(BigOrSmall, 2);
const BigOrSmall = union(enum) { small: u8, big: u128 };
```

In `replica_test.zig` `peer_path()`, there is a partial workaround by looping over pointers:

```zig
for (peers.constSlice()) |*process_b| {
```

but as the original issue demonstrates, it only helps in `Debug` mode.

**Comment by bkmgit** [comment]
Thanks.
